### PR TITLE
test/vulkan: fix/workaround validation errors

### PR DESCRIPTION
**Pull request by djdeath**
Fix the vulkan test application behavior with resizing.

## Description
Vulkan provides validation layers to help the application figure out potential programming error.
There are a number reported when using the test application with resizing enabled.
This fixes those issues. There is also a Intel/Mesa driver issue that needs to be fixed https://gitlab.freedesktop.org/mesa/mesa/-/issues/5177

This is addressing https://github.com/libsdl-org/SDL/issues/4595

## Existing Issue(s)
There are a number reported when using the test application with resizing enabled.

### CODE REVIEW

File: test/testvulkan.c [hunk 1]
```diff
@@ -1176,7 +1184,7 @@ int main(int argc, char *argv[])
     {
         SDL_Log("%2.2f frames per second\n", ((double)frames * 1000) / (now - then));
     }
-    quit(0);
+    shutdownVulkan(SDL_FALSE);
```
Comment (sezero) [inline test/testvulkan.c:1187]: This misses `SDLTest_CommonQuit(state);` ?

Comment (djdeath) [reply test/testvulkan.c:1187]: Thanks, updated.

**Comment by sezero** [comment]
No one reviewed this  - @icculus, @slouken? Needs rebasing, btw..

**Comment by sulix** [comment]
FWIW, I started to test this back when it first appeared, but kept hitting the Intel/Mesa issue and gave up. Grabbing the latest version and testing it still has the crash issues on Intel, but definitely fixes the validation errors, and works fine on llvmpipe.

However, there's still one validation error that appears consistently upon closing the window:
```
VUID-vkDestroySurfaceKHR-surface-01266(ERROR / SPEC): msgNum: -1093400348 - Validation Error: [ VUID-vkDestroySurfaceKHR-surface-01266 ] Object 0: handle = 0xeee7f0, type = VK_OBJECT_TYPE_INSTANCE; | MessageID = 0xbed408e4 | vkDestroySurfaceKHR() called before its associated VkSwapchainKHR was destroyed. The Vulkan spec states: All VkSwapchainKHR objects created for surface must have been destroyed prior to destroying surface (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkDestroySurfaceKHR-surface-01266)
```

Changing that last ``SDL_FALSE`` in the ``shutdownVulkan()`` call at the end of ``main()`` to ``SDL_TRUE`` seems to fix that, and hasn't caused any other issues for me.

**Comment by icculus** [comment]
> Changing that last SDL_FALSE in the shutdownVulkan() call at the end of main() to SDL_TRUE seems to fix that, and hasn't caused any other issues for me.

There are several things that can cause SDLTest_CommonEvent() to set the `done` flag: an SDL_QUIT event by closing all existing windows, pressing ESC, or an SDL_QUIT event from a SIGINT to the process on Unix.

In some of these cases, we still have valid windows and swapchains when we hit that shutdownVulkan call, in others we do not.

We should be checking if the swapchain is already VK_NULL_HANDLE when deciding to destroy it and always destroy it if not (Which is what destroySwapchain() already does)...this patch cleaning up the swapchain before SDLTest destroys the SDL_Window is a correct fix, but (I don't think...?) the boolean argument added to shutdownVulkan is necessary here.

**Comment by icculus** [comment]
ok, I'm merging this.

**Pull request merged**